### PR TITLE
chore(depschecker): check and resolve dependencies in localEnvManager

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Check / resolve .net / func / ngrok
Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/12863541/

### CODE REVIEW

File: packages/fx-core/src/common/local/localEnvManager.ts [hunk 1]
```diff
@@ -105,4 +133,28 @@ export class LocalEnvManager {
...
+    if (ProjectSettingsHelper.includeBackend(projectSettings)) {
+      depsTypes.push(DepsType.FunctionNode);
+      depsTypes.push(DepsType.FuncCoreTools);
+    }
+
+    if (ProjectSettingsHelper.includeBot(projectSettings)) {
+      depsTypes.push(DepsType.Ngrok);
+    }
+
+    if (ProjectSettingsHelper.isSpfx(projectSettings)) {
```
Comment (a1exwang) [inline packages/fx-core/src/common/local/localEnvManager.ts:154]: From my understanding, for Azure project, only AzureNode checker is needed and for SPFx projects, only SpfxNode checker is needed.

Comment (XiaofuHuang) [reply packages/fx-core/src/common/local/localEnvManager.ts]: Remove node checker and will implement it in another PR.

**Review by a1exwang — APPROVED** [review]

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/src/common/deps-checker/depsLogger.ts [hunk 2]
```diff
@@ -18,3 +19,59 @@ export interface DepsLogger {
 
   cleanup(): void;
 }
+
+// TODO: Implement DepsLoggerAdapter
+export class DepsLoggerAdapter implements DepsLogger {
```
[the remark this answers is not in the document]
Comment (XiaofuHuang) [reply packages/fx-core/src/common/deps-checker/depsLogger.ts:24]: I think that after implementing the command `teamsfx doctor`,  the dependency installation result will be returned to cli / vsc. Then cli / vsc can decide how to print out the result. 
The adapter is used to keep the logic can work during `teamsfx doctor` development.

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/src/common/deps-checker/depsLogger.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (swatDong) [reply packages/fx-core/src/common/deps-checker/depsLogger.ts]: (Could consider in following PRs) How about implementing `DepsLogger` as a class in core? Just let `DepsLogger` to use `core.logger`, then no adapter is needed (also no vsc/cli adapters are needed).
@meifans - what's your plan on implementing `DepsLogger` during refactoring?

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

File: packages/fx-core/src/common/deps-checker/depsLogger.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (meifans) [reply packages/fx-core/src/common/deps-checker/depsLogger.ts]: Deps Checker is independent with Fx.core init() and it's lifecycle, so not use fx.core logger.
Long term, there should be only one deps logger used by the logic in Fx-core.
I will co-work with @XiaofuHuang on it.

**Review by meifans — APPROVED** [review]

**Pull request merged**